src/libs/karm-text/prose.h

```diff
@@ -204,6 +206,8 @@ struct Prose : Meta::Pinned {
             _currentSpan = _currentSpan->parent;
     }
 
+    void overrideSpanStackWith(Prose const& prose);
+
```

Comment:
Abstract the span tree into a separate class, which will own the spans (RC).
Each prose will still have a the list of spans used in it.
Spans ended before a prose A is interrupted to prose B should not be in A.

Reply:
I didn't see a lot of impact in moving it to its own class now